rsql-jpa/src/test/java/io/github/perplexhub/rsql/JoinUtilsTest.java

```diff
@@ -0,0 +1,132 @@
+package io.github.perplexhub.rsql;
+
+import jakarta.persistence.criteria.Fetch;
+import jakarta.persistence.criteria.From;
+import jakarta.persistence.criteria.Join;
+import jakarta.persistence.criteria.JoinType;
+import jakarta.persistence.metamodel.Attribute;
+import org.hibernate.metamodel.model.domain.SingularPersistentAttribute;
```

Comment:
I had to use different internal Hibernate 6.x classes to test that we can get a join from fetches.